zerver/views/user_groups.py

```diff
@@ -138,12 +138,20 @@ def is_api_value_same_as_current_setting_value(
     ) == set(current_value["direct_subgroups"])
 
 
-def check_setting_value_changed(
+def can_edit_group_setting(
     current_value: UserGroup,
     new_setting_value: Union[int, AnonymousSettingGroupDict],
+    previous_setting_value: Optional[Union[int, AnonymousSettingGroupDict]],
 ) -> bool:
     parsed_current_setting_value = get_parsed_group_setting_value(current_value)
 
+    if previous_setting_value is not None and not is_api_value_same_as_current_setting_value(
+        parsed_current_setting_value, previous_setting_value
+    ):
+        # We do not allow changing the setting if the value sent in
+        # "_previous" field is not same as the current setting value.
+        raise PreviousSettingValueMismatchedError
```

Comment:
Like with the API documentation, this is a good place for a comment, but the should hint concisely what's going on -- e.g., that this check is here to help prevent races, by refusing to change a setting where the client (and thus UI it was presenting to the user) showed a different existing state.